src/core/thread/mle.cpp

```diff
@@ -788,15 +788,20 @@ ThreadError Mle::AppendLeaderData(Message &aMessage)
     return aMessage.Append(&mLeaderData, sizeof(mLeaderData));
 }
 
+void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
```

Comment:
Okay, I think I've caught up to your thinking now.  It's fine to keep the `Fill*Tlv` methods for now.  We can work on unifying the TLV implementations in a later PR.  Let me review again.